mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterScalerActor.java

```diff
@@ -323,24 +363,37 @@ static class ClusterAvailabilityRule {
         private final ResourceClusterScaleSpec scaleSpec;
         private final Clock clock;
         private Instant lastActionInstant;
+        private boolean enabled;
 
         public ClusterAvailabilityRule(ResourceClusterScaleSpec scaleSpec, Clock clock) {
             this.scaleSpec = scaleSpec;
             this.clock = clock;
 
             this.lastActionInstant = Instant.MIN;
+            this.enabled = true;
         }
-        public Optional<ScaleDecision> apply(UsageByGroupKey usage) {
-            // Cool down check
-            if (this.lastActionInstant.plusSeconds(this.scaleSpec.getCoolDownSecs()).compareTo(clock.instant()) > 0) {
-                log.debug("Scale CoolDown skip: {}, {}", this.scaleSpec.getClusterId(), this.scaleSpec.getSkuId());
-                return Optional.empty();
-            }
 
-            this.lastActionInstant = clock.instant();
+        public void resetLastActionInstant() {
```

Comment:
nit: I would update the ticker from within the DS whenever any field of the DS is being modified. This way, you don't have to make explicit calls for every interaction. Also, in the future, when you have new references, there's less chance of making a mistake.